controllers/snapshot/snapshot_adapter.go

```diff
@@ -148,6 +148,13 @@ func (a *Adapter) EnsureRerunPipelineRunsExist() (controller.OperationResult, er
 
 		pipelineRun, err := a.createIntegrationPipelineRun(a.application, integrationTestScenario, a.snapshot)
 		if err != nil {
+			if clienterrors.IsInvalid(err) {
+				a.logger.Error(err, "pipelineRun failed during creation due to invalid resource")
+				testStatuses.UpdateTestStatusIfChanged(
```

Comment:
`gitops.WriteIntegrationTestStatusesIntoSnapshot（）` should be called as well after this function to write the updated status to snapshot annotation.

Reply:
Has been updated [here](https://github.com/redhat-appstudio/integration-service/pull/509/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R159)